etcd-backup/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/spf13/cobra"
+	"go.etcd.io/etcd/client/pkg/v3/transport"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/client/v3/snapshot"
+
+	"go.uber.org/zap"
+)
+
+const (
+	DefaultDialTimeout       = 15 * time.Second
+	DefaultEtcdClientTimeout = 5 * time.Minute
+)
+
+type options struct {
+	clusterID      string
+	etcdEndpoint   string
+	s3BucketName   string
+	s3BucketRegion string
+}
+
+func main() {
+	var opts options
+
+	cmd := &cobra.Command{
+		Use:          "etcd-backup",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(opts)
```

Comment:
Would it make sense to propagate context here to permit graceful termination on SIGINT?

Reply:
which context are you referring to?

Reply:
`cmd.Context()` see https://pkg.go.dev/github.com/spf13/cobra#Command.Context. Typically you would initialize a context with https://pkg.go.dev/os/signal#NotifyContext and pass that to https://pkg.go.dev/github.com/spf13/cobra#Command.ExecuteContext. Then whenever you retrieve context from the command it will be marked Done on SIGINT which will allow any functions you've passed context to the chance to gracefully exit.

Reply:
I see, thanks @Ajpantuso, I implemented your suggestion.